core/tests/puppeteer-acceptance-tests/utility-functions/blogPostAdminUtils.js

```diff
@@ -0,0 +1,132 @@
+    
+    await console.log("Successfully created a draft blog post!");
+    await this.browserInstance.goto(blogDashboardUrl);
+
+    // this.page = await this.browserInstance.browser.newPage();
+  }
+
+  async deleteDraftBlogPost() {
+    await (this.browserInstance).clickOn("span", "Delete", 100);
+    await (this.browserInstance).clickOn("button", " Confirm ");
+
+    console.log("draft blog post with given title deleted successfully!");
+  }
+
+  async deleteDraftBlogPostByTitle(draftBlogPostTitle) {
+    let draftBlogPostInfo = {
+      "draftBlogPostTitle": draftBlogPostTitle,
+      "deleteDraftBlogPost": this.deleteDraftBlogPost(),
+    }
+    await (this.page).evaluate(async(draftBlogPostInfo) => {
+      const allDraftBlogPosts = document.getElementsByClassName('blog-dashboard-tile-content');
+      for(let i = 0; i < allDraftBlogPosts.length; i++) {
+        let draftBlogPostTitle = allDraftBlogPosts[i].getElementsByClassName('e2e-test-blog-post-title')[0].innerText;
+        if(draftBlogPostTitle === draftBlogPostInfo.draftBlogPostTitle) {
+          allDraftBlogPosts[i].getElementsByClassName('e2e-test-blog-post-edit-box')[0].click();
+          await draftBlogPostInfo.deleteDraftBlogPost;
+          return;
+        }
+      }
+    }, draftBlogPostInfo);
+  }
+
+  async createNewBlogPostByTitle(newBlogPostTitle) {
+    try{
+      await (this.browserInstance).clickOn("span", "NEW POST");
+    } catch {
+      await (this.browserInstance).clickOn("span", " CREATE NEW BLOG POST ");
+    }
+    await (this.browserInstance).type(blogTitleInput, newBlogPostTitle);
+    await (this.browserInstance).type(blogBodyInput, "blog post test body content");
+
+    await (this.browserInstance).clickOn("div", thumbnailPhotoBox);
+    await (this.browserInstance).uploadFile('collection.svg');
+    await (this.browserInstance).clickOn("button", " Add Thumbnail Image ");
+    await (this.page).waitForTimeout(500);
+
+    await (this.browserInstance).clickOn("span", " International ");
+    await (this.browserInstance).clickOn("span", " DONE ");
+  }
+
+  async publishNewBlogPost() {
+    await (this.browserInstance).clickOn("span", "PUBLISH");
+    await (this.browserInstance).clickOn("button", " Confirm ", 1000);
+    
+    console.log("Successfully published a blog!");
+    await this.goto(blogDashboardUrl);
+  }
+
+  async unpublishBlogPost() {
+    await (this.browserInstance).clickOn("span", "Unpublish", 100);
+    await (this.browserInstance).clickOn("button", " Confirm ", 100);
+
+    console.log("published blog post with given title unpublished successfully!");
+  }
+
+  async unpublishBlogPostByTitle(publishedBlogPostTitle) {
+    await (this.browserInstance).clickOn("div", " PUBLISHED ");
+    await (this.page).waitForTimeout(500);
+    let publishedBlogPostInfo = {
+      "publishedBlogPostTitle": publishedBlogPostTitle,
+      "unpublishBlogPost": this.unpublishBlogPost(),
```

Comment:
This is an odd formulation because when constructing the dict you are actually calling unpublishBlogPost() I suggest that you don't have a dict at all here, and just do it directly. If you need a reference to "this" inside the inner function, then define something like `let that = this` outside it, and then use `that` when you need it.

Reply:
we can only pass 1 object inside that page.evaluate function for doing some DOM manipulations.. so I need to create a dict that I can simply pass into the function and can use many variables!

Reply:
I'm not sure I believe that. The documentation says "args" which implies multiple args are possible. Also this SO page suggests that it is possible: https://stackoverflow.com/questions/46088351/how-can-i-pass-variable-into-an-evaluate-function/52012664#52012664

Could you please check again?

Reply:
@seanlip the implementation was using page.exposeFunction().. we can call functions inside the evaluate function using the window object

Reply:
I don't understand how your response relates to my question -- could you please try explaining again, in more detail?